loky/backend/context.py

```diff
@@ -154,16 +154,46 @@ def _cpu_count_cgroup(os_cpu_count):
             return os_cpu_count
 
 
-def _cpu_count_user(os_cpu_count):
-    """Number of user defined available CPUs"""
+def _cpu_count_affinity(os_cpu_count):
     # Number of available CPUs given affinity settings
-    cpu_count_affinity = os_cpu_count
     if hasattr(os, 'sched_getaffinity'):
         try:
-            cpu_count_affinity = len(os.sched_getaffinity(0))
+            return len(os.sched_getaffinity(0))
         except NotImplementedError:
             pass
 
+    # On PyPy and possibly other platforms, os.sched_getaffinity does not exist
+    # or raises NotImplementedError, let's try with the psutil if installed.
+    try:
+        import psutil
+
+        p = psutil.Process()
+        if hasattr(p, "cpu_affinity"):
+            return len(p.cpu_affinity())
+
+    except ImportError:  # pragma: no cover
+        if (
+            sys.platform == "linux"
+            and os.environ.get('LOKY_MAX_CPU_COUNT') is None
+        ):
+            # PyPy does not implement os.sched_getaffinity on Linux which
+            # can cause severe oversubscription problems. Better warn the
+            # user in this particularly pathological case which can wreck
+            # havoc, typically on CI workers.
+            warnings.warn(
+                "Failed to inspect CPU affinity constraints on this system. "
+                "Please install psutil or explictly set LOKY_MAX_CPU_COUNT."
+            )
+
+    # This can happen for platforms that do not implement any kind of CPU
+    # infinity such as macOS-based platforms.
+    return os_cpu_count
```

Comment:
I don't understand why this line is not covered by the tests... The macOS workers on the CI should cover this...

Reply:
I checked that the macOS CI did run:

```
tests/test_loky_module.py::test_cpu_count PASSED                         [ 17%]
```

which should have covered this line.

I tested with a debug commit to raise `ValueError` and it did raise on macOS. So there is definitely a problem with the aggregation of the coverage data but life is too short.